python/mxnet/gluon/nn/conv_layers.py

```diff
@@ -1007,3 +1008,37 @@ def __init__(self, layout='NCDHW', **kwargs):
         assert layout == 'NCDHW', "Only supports NCDHW layout for now"
         super(GlobalAvgPool3D, self).__init__(
             (1, 1, 1), None, 0, True, True, 'avg', **kwargs)
+
+
+class ReflectionPad2D(HybridBlock):
+    """Pads the input tensor using the reflection of the input boundary.
+
+    Parameters
+    ----------
+    padding: int or a tuple of int
+        An integer padding width for height and weight or a tuple of
+        integers padding widths for each axis of the format ``(before_1, after_1, ... ,
+        before_N, after_N)``. The `padding` should be of length ``2*N`` where ``N``
```

Comment:
The document describes something general but the class is specific to 2D. Maybe describe the argument that's specific to 2D case here.

Reply:
If pass into a tuple, it works for different dimensions. Should we remove this?

Reply:
Since the class name says 2D, it should support 2D.

You can do something similar to the conv blocks, with a common abstract class for the forward logic, and the actual 1D/2D/3D classes for specific implementation.

Reply:
Agree. Just made the changes. Will refactor when adding 1D/3D